crates/biome_fs/src/path.rs

```diff
@@ -149,11 +149,22 @@ impl BiomePath {
     pub fn is_dir(&self) -> bool {
         matches!(self.kind, FileKinds::Directory)
     }
+
     #[inline(always)]
     pub fn is_handleable(&self) -> bool {
         matches!(self.kind, FileKinds::Handleable)
     }
 
+    /// Returns `true` for file types that must be scanned (if the scanner is
+    /// enabled) because Biome's own functionality relies on them.
+    #[inline(always)]
+    pub fn is_required_during_scan(&self) -> bool {
+        matches!(
+            self.kind,
+            FileKinds::Config | FileKinds::Ignore | FileKinds::Manifest
```

Comment:
Are we intentionally adding `tsconfig.json` too?

Reply:
Yes, because it's required for resolving imports in projects. Only inside `node_modules` we exclude them, because they shouldn't affect published packages (ideally, published packages don't even have them, but well... NPM).